DoctorWho.Web/Controllers/DoctorsController.cs

```diff
@@ -31,11 +37,40 @@ public ActionResult<IEnumerable<DoctorResource>> GetDoctors()
             var doctorsResources = mapper.Map<IEnumerable<DoctorResource>>(doctors);
             return Ok(doctorsResources);
         }
-        //public async Task<ActionResult<IEnumerable<DoctorResource>>> GetDoctors()
-        //{
-        //    var doctors = await doctorService.ListAsync();
-        //    var doctorsResources = mapper.Map<IEnumerable<Doctor>, IEnumerable<DoctorResource>>(doctors);
-        //    return Ok(doctorsResources);
-        //}
+
+        [HttpGet("{doctorId}", Name = "GetDoctor")]
+        public async Task<ActionResult<DoctorResource>> GetDoctor(int doctorId)
+        {
+            var doctor = await doctorService.GetDoctorAsync(doctorId);
+            if (doctor == null)
+            {
+                return NotFound();
+            }
+            var doctorResource = mapper.Map<DoctorResource>(doctor);
+            return Ok(doctorResource);
+        }
+
+        [HttpPut("{doctorId}")]
+        public async Task<IActionResult> UpdateOrCreateDoctor(int doctorId, DoctorManipulationResource doctor)
+        {
+            var doctorToUpdate = await doctorService.GetDoctorAsync(doctorId);
+            var doctorValidator = new DoctorManipulationResourceValidator();
+            ValidationResult result = doctorValidator.Validate(doctor);
+            if(!result.IsValid)
+            {
+                return BadRequest(result.ToString());
+            }
+            if (doctorToUpdate == null)  
+            { //upsert
```

Comment:
when you find yourself need to add such comment to explain something consider taking it to it's meaningfull-named method.